Authentication + hotfix

Maxou974 opened the pull request:
Merged authentication backend with hotfix branch, few modifications on Makefile (tell me what you think), need to test it on 42 computer since I have done this on my PC, but once u run make, the front, back and postgres container will be launched and request can be made to them.

Bee-Have committed:
✨ -simple dockerfile mockup

Bee-Have committed:
✨🚧 -new simple docker compose and temp react empty app. HOWEVER it has not been tested yet

Bee-Have committed:
🙈 -updated gitignore since there is a ".env" file i wish to keep. This ".env" file will have paths and non sensitive informations (no passwords)

Bee-Have committed:
✨ -added simple makefile to help testing

Bee-Have committed:
🐛 -error in makefile fclean rule

Bee-Have committed:
🙈 -added personnal notes to gitignore to not push them by accident

Bee-Have committed:
✅🐛 -working simple react app front

Maxou974 committed:
authentication is working

Maxou974 committed:
authentication is working good

Maxou974 committed:
adding gitignore and removed ignored file

Maxou974 committed:
adding 2FA

Maxou974 committed:
little fix

shamizi committed:
adding getUserProfile route

Maxou974 committed:
adding username unique to schema.prisma

shamizi committed:
chatProfilDTO

shamizi committed:
Merge branch 'authentication' of https://github.com/Bee-Have/ft_transcendence into authentication

Maxou974 committed:
adding get image route

Maxou974 committed:
git add *

Maxou974 committed:
merge

Maxou974 committed:
adding a bit of swagger

Maxou974 committed:
addding more swagger

Commit:
Hotfix Makefile / error language / etc

- Changed JS files for Typescript ones.
- Updated .gitignore to not include node_modules.
- Makefile should properly work now.
- All 3 containers are written, now should link them and WIP on the backend.

Maxou974 committed:
adding more swagger

Maxou974 committed:
adding more swagger for the user route

Maxou974 committed:
hmhmmhh

Maxou974 committed:
update readme

Maxou974 committed:
prettier readme

Maxou974 committed:
fucked up the last readme

Commit:
renaming back_end in back

Commit:
merging with htofix

Commit:
new dir

Commit:
trying to merge and docker

Commit:
Fixing some issues from Makefile and Docker-compose

- Fixed issue with enter_nginx by changing into enter_back_end
- Changed port usage in Docker-compose.yml to use the .env file

Maxou974 committed:
back_end dockerize

Maxou974 committed:
gitignore update

Maxou974 committed:
merge hotfix

Maxou974 committed:
adding .env.example

Maxou974 committed:
back_end is docker-composed, back_end hot reload, back_end connected with db

Maxou974 committed:
fixing the state variable for 42 auth

Kopeerde commented:
There is a ton of files and way too many lines for me to check perfectly everything but it seems to work well, i'll quickly send 2/3 changes like finishing your foreign key implementation and delete some files.

Commit:
Finishing foreign keys and cleanup

- Finished your implementation of foreign keys for your Channel Model and User Model.
- Removed depedency to Bcrypt since you changed in favor to Argon2.
- Added a .gitignore rule to ignore any .jpeg uploaded as avatar.
